File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/MainContract.java
package com.remotearthsolutions.expensetracker.contracts;

public interface MainContract {

    interface View {
        void initializeView();

        void goBackToLoginScreen();

        void onLogoutSuccess();

        void startLoadingApp();

        void showTotalExpense(String amount);

        void showTotalBalance(String amount);

        void stayOnCurrencyScreen();

        void setBalanceTextColor(int colorId);
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/MainContract.kt
package com.remotearthsolutions.expensetracker.contracts

interface MainContract {
    interface View : BaseView {
        fun initializeView()
        fun goBackToLoginScreen()
        fun onLogoutSuccess()
        fun startLoadingApp()
        fun showAllTimeTotalExpense(amount: String?)
        fun showTotalBalance(amount: String?)
        fun stayOnCurrencyScreen()
        fun setBalanceTextColor(colorId: Int)
        fun onDataUpdated()
    }
}